### PR TITLE
Additional tests for `polyfill`

### PR DESCRIPTION
**Pull request by isaacbrodsky**
There are some tests in this PR which we currently expect to fail.

### CODE REVIEW

File: src/apps/applib/lib/utility.c [hunk 1]
```diff
@@ -233,3 +233,16 @@ void randomGeo(GeoCoord* g) {
         (((float)rand() / (float)(RAND_MAX)) * 180.0) - 90.0);
     g->lon = H3_EXPORT(degsToRads)((float)rand() / (float)(RAND_MAX)) * 360.0;
 }
+
+/**
+ * Returns the number of non-invalid indexes in the array.
+ */
+int countActualHexagons(H3Index* hexagons, int numHexagons) {
```
Comment (nrabinowitz) [inline src/apps/applib/lib/utility.c]: Would `countValidHexagons` be more accurate?

File: src/apps/testapps/testPolyfill.c [hunk 2]
```diff
@@ -41,14 +42,68 @@ static GeoCoord emptyVerts[] = {{0.659966917655, -2.1364398519394},
...
+}
+
+static void fillIndex_assertions(H3Index h) {
+    if (isTransmeridianCell(h)) {
+        // TODO: these do not work correctly
+        return;
+    }
+
+    int currentRes = H3_EXPORT(h3GetResolution)(h);
+    for (int nextRes = currentRes; nextRes <= currentRes + 1; nextRes++) {
```
Comment (nrabinowitz) [inline src/apps/testapps/testPolyfill.c]: This only goes down one resolution, right? Should `1` be a clearly-named const like `testDepth`? If not, should this be a loop at all?

File: src/apps/testapps/testPolyfill.c [hunk 3]
```diff
@@ -41,14 +42,68 @@ static GeoCoord emptyVerts[] = {{0.659966917655, -2.1364398519394},
...
+        if (bndry.verts[i].lon < minLon) minLon = bndry.verts[i].lon;
+        if (bndry.verts[i].lon > maxLon) maxLon = bndry.verts[i].lon;
+    }
+
+    return maxLon - minLon > M_PI - (M_PI / 4);
+}
+
+static void fillIndex_assertions(H3Index h) {
+    if (isTransmeridianCell(h)) {
+        // TODO: these do not work correctly
```
Comment (nrabinowitz) [inline src/apps/testapps/testPolyfill.c]: Is this true for all transmeridian cells, or just polar cells? I thought transmeridian cells in general should work correctly.

Comment (isaacbrodsky) [reply src/apps/testapps/testPolyfill.c]: I recall this was an issue for not only the cells at the poles.

Comment (nrabinowitz) [reply src/apps/testapps/testPolyfill.c]: If so, that's a problem. The `polyfill` algo is supposed to work correctly for transmeridian polygons, and we should have tests to confirm.

File: src/apps/applib/lib/utility.c (hunk 1 above)
Comment (isaacbrodsky) [reply src/apps/applib/lib/utility.c]: They're not necessarily validated with `h3IsValid` - just checked for not being 0, which I take to mean "null" or "missing".

File: src/apps/testapps/testPolyfill.c (hunk 3 above)
Comment (isaacbrodsky) [reply src/apps/testapps/testPolyfill.c]: I tried replacing `isTransmeridianCell` with this body, and the test suite failed:
```
    int res = H3_GET_RESOLUTION(h);
    GeoCoord northPoleCoords = {.lat = M_PI_2, .lon = 0};
    H3Index northPoleIndex = H3_EXPORT(geoToH3)(&northPoleCoords, res);
    if (northPoleIndex == h) {
        return true;
    }
    GeoCoord southPoleCoords = {.lat = -M_PI_2, .lon = 0};
    H3Index southPoleIndex = H3_EXPORT(geoToH3)(&southPoleCoords, res);
    if (southPoleIndex == h) {
        return true;
    }
    return false;
```

File: src/apps/testapps/testPolyfill.c (hunk 2 above)
Comment (isaacbrodsky) [reply src/apps/testapps/testPolyfill.c]: I didn't test 2 (or more) because the assertion fails. Added a comment noting this.

**Pull request merged**